gradle/util.gradle

```diff
@@ -0,0 +1,55 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Util functions to resolve configuration values from runtime/gradle/environment properties/variables
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Try to get configuration property {@code propertyName} in the next order:<ol>
+ *     <li>-D${propertyName} Java runtime property in gradle command</li>
+ *     <li>-P${propertyName} gradle runtime property in gradle command</li>
+ *     <li>${propertyName} in ~/.gradle/gradle.properties (user global gradle properties)</li>
+ *     <li>${propertyName} in gradle.properties (project local gradle properties)</li>
```

Comment:
I think you got the order wrong/confused:
1. https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties
2. https://discuss.gradle.org/t/order-of-precedence-for-defining-gradle-properties/7274

Also, It's stated that 
> (if an option is configured in multiple locations the last one wins 

I think this is worth to mention here.

Reply:
1. i've manually verified the order, so i'm not sure which is wrong/confusing
2. The first one wins: 1adba1b

Reply:
ah ok, I think from the explanation in the gradle docs I understand now, so configurations are applied in the order in the docs, and the last ones override the ones before. So basically the precedence will be in reverse order.